widget/playground/src/components/TokensPanel/TokensPanel.List.tsx

```diff
@@ -0,0 +1,195 @@
+import type { TokensListProps, TokenType } from './TokensPanel.types';
+
+import {
+  Checkbox,
+  CloseIcon,
+  Divider,
+  IconButton,
+  Image,
+  ListItemButton,
+  NotFound,
+  SearchIcon,
+  Switch,
+  TextField,
+  Typography,
+  VirtualizedList,
+} from '@rango-dev/ui';
+import React, { useEffect, useLayoutEffect, useState } from 'react';
+
+import {
+  CheckList,
+  IconWrapper,
+  SelectButton,
+  SelectDeselectText,
+} from '../MultiList/MultiList.styles';
+import { InnerElementType } from '../SingleList';
+import { EmptyContainer, StyledButton } from '../SingleList/SingleList.styles';
+
+import { TokensHeaderList } from './TokensPanel.styles';
+
+const PAGE_SIZE = 20;
+const ITEM_HEIGHT = 46;
+export function TokensList(props: TokensListProps) {
+  const {
+    list,
+    onChange,
+    onChangeAll,
+    onConfirm,
+    showSelectedTokens,
+    setShowSelectedTokens,
+    isAllSelected,
+  } = props;
+  const [searchValue, setSearchValue] = useState('');
+  const [virtualList, setVirtualList] = useState(list);
+  const [hasNextPage, setHasNextPage] = useState(true);
+
+  // Filter tokens based on search criteria
+  const filteredTokens = searchValue
+    ? list.filter((token) =>
+        token.symbol.toLowerCase().includes(searchValue.toLowerCase())
+      )
+    : list;
+
+  // Tokens to display, either all or selected
+  const tokensToDisplay = showSelectedTokens
+    ? filteredTokens.filter((token) => token.checked)
+    : filteredTokens;
+
+  useEffect(() => {
+    setHasNextPage(tokensToDisplay.length > virtualList.length);
```

Comment:
[Screencast from 23-11-01 15:43:09.webm](https://github.com/rango-exchange/rango-client/assets/120931936/bcbccb25-009e-4066-83e4-ef17c9af2046)
Some of the tokens are unselected state, but it still shows that all the tokens are selected

Reply:
fixed.